pkg/schedule/schedule.go

```diff
@@ -149,17 +179,29 @@ func (f *fifo) run() {
 				f.mu.Unlock()
 				// clean up pending jobs
 				for _, todo := range pendings {
-					todo(f.ctx)
+					f.executeJob(todo, true)
 				}
 				return
 			}
 		} else {
-			todo(f.ctx)
+			f.executeJob(todo, false)
+		}
+	}
+}
+
+func (f *fifo) executeJob(todo Job, updatedFinishedStats bool) {
+	defer func() {
+		if !updatedFinishedStats {
 			f.finishCond.L.Lock()
 			f.finished++
 			f.pendings = f.pendings[1:]
 			f.finishCond.Broadcast()
 			f.finishCond.L.Unlock()
 		}
-	}
+		if err := recover(); err != nil {
+			f.lg.Panic("execute job failed", zap.String("job", todo.Name()), zap.Any("err", err))
```

Comment:
Have you tested how this log will look like?

Reply:
![image](https://user-images.githubusercontent.com/21985684/173822126-f953e602-c1a0-48ae-b2df-44839c08a3d6.png)
 the `err` can't use the zap.Error(), because the `zap.Error()` need a param of the `error` type. In this code, the type of the `err` is `Any`.

Reply:
ack, then maybe we shouldn't use `err`. Using it could potentially pass different type under, maybe instead use `panic` key?

Reply:
Good idea. I have changed it.